pkg/broker/ingress/handler_test.go

```diff
@@ -1,72 +1,143 @@
 
-// TestHandlerChannelInbound uses a channel based mock client for both inbound and decouple.
-func TestHandlerChannelInbound(t *testing.T) {
-	inbound, ic := test.NewMockReceiverClient(t, 1)
-	decouple, dc := test.NewMockSenderClient(t, 1)
-	_, cleanup := createAndStartIngress(t, inbound, decouple)
-	defer cleanup()
+var topic = "topic1"
 
-	input := createTestEvent()
-	// Send an event to the inbound receiver client.
-	ic <- input
-	// Retrieve the event from the decouple sink.
-	output := <-dc
+var brokerConfig = &config.TargetsConfig{
+	Brokers: map[string]*config.Broker{
+		"ns1/broker1": {
+			Id:        "b-uid-1",
+			Name:      "broker1",
+			Namespace: "ns1",
+			DecoupleQueue: &config.Queue{
+				Topic: topic,
+			},
+		},
+	},
+}
 
-	if dif := cmp.Diff(input, output); dif != "" {
-		t.Errorf("Output event doesn't match input, dif: %v", dif)
-	}
+type testCase struct {
+	name    string
+	options []HandlerOption
+	// If port is different than the default(8080), a matching WithPort option should be provided.
+	port int
+	// in happy case, path should match the /<ns>/<broker> in the brokerConfig.
+	path  string
+	event *cloudevents.Event
+	// If method is empty, POST will be used as default.
+	method string
+	// body and header can be specified if the client is making raw HTTP request instead of via cloudevents.
+	body     map[string]string
+	header   nethttp.Header
+	wantCode int
 }
 
-// TestHandlerDefaultHTTPInbound uses default HTTP inbound client and a channel based mock decouple
-// sink.
-func TestHandlerDefaultHTTPInbound(t *testing.T) {
-	decouple, dc := test.NewMockSenderClient(t, 1)
-	_, cleanup := createAndStartIngress(t, nil, decouple)
-	defer cleanup()
+func TestHandler(t *testing.T) {
+	tests := []testCase{
+		{
+			name:     "happy case",
+			path:     "/ns1/broker1",
+			event:    createTestEvent("test-event"),
+			port:     defaultPort,
+			wantCode: nethttp.StatusOK,
+		},
+		{
+			name:     "with custom port number",
+			path:     "/ns1/broker1",
+			event:    createTestEvent("test-event"),
+			port:     8081,
+			options:  []HandlerOption{WithPort(8081)},
+			wantCode: nethttp.StatusOK,
+		},
+		{
+			name:     "valid event but unsupported http  method",
+			method:   "PUT",
+			path:     "/ns1/broker1",
+			event:    createTestEvent("test-event"),
+			port:     defaultPort,
+			wantCode: nethttp.StatusMethodNotAllowed,
+		},
+		{
+			name:     "malformed path",
+			path:     "/ns1/broker1/and/something/else",
+			event:    createTestEvent("test-event"),
+			port:     defaultPort,
+			wantCode: nethttp.StatusNotFound,
+		},
+		{
+			name:     "request is not an event",
+			path:     "/ns1/broker1",
+			port:     defaultPort,
+			wantCode: nethttp.StatusBadRequest,
+			header:   nethttp.Header{},
+		},
+		{
+			name:     "wrong path - broker doesn't exist in config",
+			path:     "/ns-1/broker2",
+			event:    createTestEvent("test-event"),
+			port:     defaultPort,
+			wantCode: nethttp.StatusInternalServerError,
```

Comment:
Shouldn't this 404?

Reply:
This one is tricky. The request can fail due to bad request, or due to the configmap volume propagation delay (which should be 500s). 

I am open to suggestion here.

Reply:
My personal preference is 404, as that is the correct steady state error in my opinion. @grantr thoughts?

Reply:
I agree it should be 404. Technically 404 may be temporary so this seems like a correct response. Same with the above broker-not-exist test.

Reply:
For the empty queue case, I am using 500s for now, since we assume the controller always fill the queue for a broker.

As Grant mentioned, we may mark the broker as `DELETED` with empty queue. If that happens, I will refactor the code to check broker status and return proper status code.